src/utils/nnls.cpp

```diff
@@ -0,0 +1,748 @@
+/**

```

Comment:
this file looks horrible, however, this is exactly what I would expect from an automatically translated Fortran code ;)

Maybe add an anonymous namespace to explicitly mark what is only needed internally?

Reply:
I tried finding sth. proper, but that's all I got ^^

PS: I added an anonymous namespace :)